sentry-tower/src/http.rs

```diff
@@ -0,0 +1,146 @@
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use http_::Request;
+use tower_layer::Layer;
+use tower_service::Service;
+
+/// Tower Layer that logs Http Request Headers.
+///
+/// The Layer can also optionally start a new performance monitoring transaction
+/// based on incoming distributed tracing headers.
+#[derive(Clone, Default)]
+pub struct SentryHttpLayer {
+    start_transaction: bool,
+}
+
+impl SentryHttpLayer {
+    /// Creates a new Layer that only logs Request Headers.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Creates a new Layer that also starts a new performance monitoring transaction.
+    pub fn with_transaction() -> Self {
+        Self {
+            start_transaction: true,
+        }
+    }
+}
+
+/// Tower Service that logs Http Request Headers.
+///
+/// The Service can also optionally start a new performance monitoring transaction
+/// based on incoming distributed tracing headers.
+#[derive(Clone)]
+pub struct SentryHttpService<S> {
+    service: S,
+    start_transaction: bool,
```

Comment:
rather minor, but if you agree with the above re-wording (or a variation) maybe calling this `continue_transaction` or so would be better as well

Reply:
Well it still does a `start_transaction`, and as commented above, the transaction itself is independent, it just belongs to a trace based on the incoming headers. Also the incoming headers are completely optional so this starts a completely new transaction if the headers are not present.